python/flink_agents/runtime/memory/chroma_long_term_memory.py

```diff
@@ -0,0 +1,456 @@
+
+    @property
+    def client(self) -> ChromaClient:
+        """Return ChromaDB client, creating it if necessary."""
+        if self.__client is None:
+            # Choose client type based on configuration
+            if self.api_key is not None:
+                # Cloud mode
+                self.__client = CloudClient(
+                    tenant=self.tenant,
+                    database=self.database,
+                    api_key=self.api_key,
+                )
+            elif self.host is not None:
+                # Client-Server Mode
+                # create database if needed
+                admin_client = chromadb.AdminClient(
+                    Settings(
+                        chroma_api_impl="chromadb.api.fastapi.FastAPI",
+                        chroma_server_host=self.host,
+                        chroma_server_http_port=self.port,
+                    )
+                )
+                try:
+                    admin_client.get_database(name=self.database, tenant=self.tenant)
+                except Exception:
+                    admin_client.create_database(name=self.database, tenant=self.tenant)
+                self.__client = chromadb.HttpClient(
+                    host=self.host,
+                    port=self.port,
+                    settings=self.client_settings,
+                    tenant=self.tenant,
+                    database=self.database,
+                )
+            else:
+                err_msg = (
+                    "As long-term memory backend, chromadb must run in cloud mode or client-server mode, "
+                    "user should provide cloud api-key or server address."
+                )
+                raise RuntimeError(err_msg)
+
+        return self.__client
+
+    @override
+    def create_memory_set(
+        self,
+        name: str,
+        item_type: type[str] | Type[ChatMessage],
+        capacity: int,
+        reduce_setup: ReduceSetup,
+    ) -> MemorySet:
+        try:
+            return self.get_memory_set(name=name)
+        except NotFoundError:
+            memory_set = MemorySet(
+                name=name,
+                item_type=item_type,
+                size=0,
+                capacity=capacity,
+                reduce_setup=reduce_setup,
+                ltm=self,
+            )
+            self.client.get_or_create_collection(
+                name=name, metadata={"serialization": memory_set.model_dump_json()}
+            )
+            return memory_set
+
+    @override
+    def get_memory_set(self, name: str) -> MemorySet:
+        metadata = self.client.get_collection(name=name).metadata
+        memory_set = MemorySet.model_validate_json(metadata["serialization"])
+        memory_set.ltm = self
+        return memory_set
+
+    @override
+    def delete_memory_set(self, name: str) -> None:
+        self.client.delete_collection(name=name)
+
+    @override
+    def add(self, memory_set: MemorySet, memory_item: str | ChatMessage) -> None:
```

Comment:
I had a feeling that adding items to long-term memory can take time, for embedding. We probably should also provide async apis.